bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/internal/client/EditingContextImpl.java

```diff
@@ -11,24 +11,28 @@
 
 package org.eclipse.emfcloud.modelserver.internal.client;
 
+import java.util.function.BiFunction;
+
 import org.eclipse.emf.common.command.Command;
 import org.eclipse.emfcloud.modelserver.client.EditingContext;
-import org.eclipse.emfcloud.modelserver.client.ModelServerClient;
 import org.eclipse.emfcloud.modelserver.command.CCommand;
 import org.eclipse.emfcloud.modelserver.common.codecs.EncodingException;
 import org.eclipse.emfcloud.modelserver.edit.CommandCodec;
 import org.eclipse.emfcloud.modelserver.edit.EMFCommandCodec;
 
+import com.fasterxml.jackson.databind.JsonNode;
+
 import okhttp3.Response;
 import okhttp3.WebSocket;
 import okhttp3.WebSocketListener;
 
 /**
  * Implementation of the client's editing context.
  */
-public class EditingContextImpl extends WebSocketListener implements EditingContext {
+public class EditingContextImpl<CLIENT> extends WebSocketListener implements EditingContext {
```

Comment:
I'm not sure. We only use the EditingContext with AbstractModelServerClient, but the EditingContext itself doesn't care which kind of client it is. As long as you give it a valid CCommand encoder, it can handle any client class.

@cdamus What do you think?

Reply:
I try to keep generic signatures as wide open as possible for clients and as narrow as possible for providers. I don't think introducing a generic type here helps the implementation in any way and it wouldn't mean anything to a user/client of the API.